src/Symfony/Component/Serializer/Normalizer/UidNormalizer.php

```diff
@@ -71,7 +71,7 @@ public function denormalize(mixed $data, string $type, ?string $format = null, a
         try {
             return $type::fromString($data);
         } catch (\InvalidArgumentException|\TypeError) {
-            throw NotNormalizableValueException::createForUnexpectedDataType(sprintf('The data is not a valid "%s" string representation.', $type), $data, [Type::BUILTIN_TYPE_STRING], $context['deserialization_path'] ?? null, true);
+            throw NotNormalizableValueException::createForUnexpectedDataType(sprintf('The data is not a valid "%s" string representation.', $type), $data, [Type::string()], $context['deserialization_path'] ?? null, true);
```

Comment:
This should not use `Type::string()` here. `NotNormalizableValueException::createForUnexpectedDataType` expects a PHP type name and we should use the string `'string'` directly instead of adding a mandatory dependency for the error handling path (btw, we have a similar issue about using `Symfony\Component\PropertyInfo\Type` right now, which is not marked as a mandatory dependency either).